services/user/Registry/service/src/utils.rs

```diff
@@ -0,0 +1,10 @@
+/// Increment the last character of a string
+///
+/// Example: "abc" -> "abd"
+pub fn increment_last_char(s: String) -> String {
+    let mut chars: Vec<char> = s.chars().collect();
+    if let Some(last) = chars.last_mut() {
+        *last = char::from_u32((*last as u32) + 1).unwrap_or(*last);
```

Comment:
Unfortunately this doesn't always work, e.g. if the last character is U+D7FF.

Reply:
Since we are restricting the chars to a-z and 0-9 the function is safe. Let me add a explicit comment about it.

Reply:
In that case, you can avoid copying the entire string by using `into_bytes`/`from_utf8`.